tartufo/scanner.py

```diff
@@ -476,3 +477,52 @@ def chunks(self):
         )
         for blob, file_path in self._iter_diff_index(diff_index):
             yield types.Chunk(blob, file_path)
+
+
+class FolderScanner(ScannerBase):
+    """Used to scan a folder."""
+
+    folder_options: types.FolderOptions
+    folder_path: str
+
+    def __init__(
+        self,
+        global_options: types.GlobalOptions,
+        folder_options: types.FolderOptions,
+        folder_path: str,
+    ) -> None:
+        """Used for scanning a folder.
+
+        :param global_options: The options provided to the top-level tartufo command
+        :param folder_options: The options specific to interacting with a folder
+        :param folder_path: The local filesystem path to scan
+        """
+        self.folder_options = folder_options
+        self.folder_path = folder_path
+        super().__init__(global_options)
+
+    @property
+    def chunks(self):
+        """Yield the individual file changes currently staged for commit.
+
+        :rtype: Generator[Chunk, None, None]
+        """
+
+        for blob, file_path in self._iter_folder():
+            if self.should_scan(file_path):
+                yield types.Chunk(blob, file_path)
+
+    def _iter_folder(self):
+        folder_path = pathlib.Path(self.folder_path)
+        for file_path in folder_path.glob(self.folder_options.pattern):
```

Comment:
By default, with your pattern of `*`, this is only going to return a top-level listing of the directory. I think there are two potential routes to consider here.

1. Use [rglob](https://docs.python.org/3/library/pathlib.html#pathlib.Path.rglob) instead, which will prepend `**/` to the pattern and automatically make this recursive
2. Consider a `--recursive` option, defaulted to `True`, and manually walk the tree.

I think both have their pros and cons. The former is definitely easier to implement, but then requires the user to fully understand glob syntax, which can be tricky for more complex situations. The latter is far more explicit and intuitive, and I think we could even get super fancy with the options doing something like `tartufo scan-folder --recursive --include-ext py --include-ext yml` and similar `--exclude-ext` options. But this is a whole big rabbit hole that would be much more complicated to implement, and most likely slower to execute as well.

On the topic of performance though, there is one other thing to note. There is a note in the [docs for glob](https://docs.python.org/3/library/pathlib.html#pathlib.Path.glob) which states:
> Note Using the “**” pattern in large directory trees may consume an inordinate amount of time.

What that "inordinate amount of time" actually means, and what alternatives in `pathlib` would be, I'm not sure. I suppose for pure raw speed in a manual walking scenario, we would want to use [os.walk](https://docs.python.org/3/library/os.html#os.walk) and yield `Path` objects only as necessary.

Reply:
Do the git scanners require `--recursive`? Glob by default enables recursion. `*` versus `**/*`. https://docs.python.org/3/library/pathlib.html#pathlib.Path.glob